File: ios/view/Pages/PlantDetailPage.swift
import SwiftUI

struct PlantDetailPage: View {
    
    @Environment(\.presentationMode) var presentation
    
    private let plant: Plant
    
    init(plant: Plant) {
        self.plant = plant
    }
    
    var body: some View {
        ZStack(alignment: .leading) {
            MaterialTheme.colors.BACKGROUND.edgesIgnoringSafeArea(.all)
            
            ScrollView() {
                VStack(alignment: .leading) {
                    
                    ZStack(alignment: .topLeading) {
                        
                        AsyncImage(urlString: plant.metadata.image)
                            .frame(height: 256)
                            .frame(
                                minWidth: 0,
                                maxWidth: .infinity
                            )
                            .clipShape(Rectangle())
                        
                        VStack {
                            
                            Spacer(minLength: 232)
                                                        
                            Image("background_curve")
                                .frame(
                                    minWidth: 0,
                                    maxWidth: .infinity,
                                    alignment: .bottom
                                )
                                .foregroundColor(MaterialTheme.colors.BACKGROUND)
                        }
                        .frame(height: 256)
                        
                        ZStack {
                            Group {
                                Image("baseline_arrow_back_black_24pt")
                                    .resizable()
                                    .frame(width: 20, height: 20)
                                    .foregroundColor(Color.white)
                            }
                            .frame(width: 32, height: 32, alignment: .center)
                            .background(MaterialTheme.colors.PRIMARY)
                            .cornerRadius(4)
                            .onTapGesture {
                                self.presentation.wrappedValue.dismiss()
                            }
                        }
                        .padding(.all, 16)
                    }
                    
                    Text(plant.metadata.title)
                        .modifier(MaterialTheme.typography.h1)
                        .padding(.leading, 24)
                        .padding(.top, 12)
                        .padding(.trailing, 24)
                        .padding(.bottom, 2)
                                        
                    Text(plant.metadata.subtitle)
                        .modifier(MaterialTheme.typography.subtitle1)
                        .padding(.leading, 24)
                        .padding(.top, 2)
                        .padding(.trailing, 24)
                        .padding(.bottom, 24)
                    
                    ContentFacts(facts: plant.facts)
                    
                    ForEach(plant.content, id: \.self) { element in
                        Group {
                            if element.type == ContentType.HEADER {
                                ContentHeader(header: element.content)
                            } else if element.type == ContentType.TEXT {
                                ContentText(text: element.content)
                            } else if element.type == ContentType.IMAGE {
                                ContentImage(url: element.content)
                            } else if element.type == ContentType.MAP {
                                ContentMap(id: element.content)
                            }
                        }
                    }
                    
                    FooterComponent()
                }
            }
        }
        .navigationBarTitle("")
        .navigationBarHidden(true)
    }
}

File: android/view/pages/PlantDetailPage.kt
package com.tobiasheymann.ep.view.pages

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.Icon
import androidx.compose.material.IconButton
import androidx.compose.material.MaterialTheme
import androidx.compose.material.Text
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.rounded.ArrowBack
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.ColorFilter
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.unit.dp
import androidx.navigation.NavController
import coil.compose.rememberImagePainter
import coil.size.Scale
import com.tobiasheymann.ep.R
import com.tobiasheymann.ep.model.ContentType
import com.tobiasheymann.ep.model.Plant
import com.tobiasheymann.ep.view.components.*

@Composable
fun PlantDetailPage(navController: NavController) {

    if (navController.currentBackStackEntry?.arguments?.containsKey("plant") == true) {
        val plant = navController.currentBackStackEntry?.arguments?.getSerializable("plant") as Plant

        LazyColumn {
            item {
                Box {
                    Image(
                        painter = rememberImagePainter(
                            data = plant.metadata.image,
                            builder = {
                                crossfade(true)
                                scale(Scale.FILL)
                                placeholder(drawableResId = R.drawable.placeholder_square_large)
                            }
                        ),
                        contentDescription = null,
                        contentScale = ContentScale.Crop,
                        modifier = Modifier
                            .fillMaxWidth()
                            .height(256.dp)
                    )
                    Image(
                        painter = painterResource(id = R.drawable.background_curve),
                        contentDescription = null,
                        modifier = Modifier
                            .fillMaxWidth()
                            .align(Alignment.BottomCenter)
                            .offset(y = 5.dp),
                        colorFilter = ColorFilter.tint(MaterialTheme.colors.background)
                    )
                    Box(modifier = Modifier.padding(16.dp)) {
                        IconButton(
                            onClick = {
                                navController.popBackStack()
                            },
                            modifier = Modifier
                                .clip(RoundedCornerShape(4.dp))
                                .background(MaterialTheme.colors.primary)
                                .size(32.dp, 32.dp),
                            content = {
                                Icon(
                                    imageVector = Icons.Rounded.ArrowBack,
                                    contentDescription = null,
                                    modifier = Modifier
                                        .size(20.dp, 20.dp),
                                    tint = Color.White
                                )
                            }
                        )
                    }
                }

                Text(
                    text = plant.metadata.title,
                    modifier = Modifier.padding(start = 24.dp, top = 12.dp, end = 24.dp, bottom = 2.dp),
                    style = MaterialTheme.typography.h1
                )
                Text(
                    text = plant.metadata.subtitle,
                    modifier = Modifier.padding(start = 24.dp, top = 2.dp, end = 24.dp, bottom = 24.dp),
                    style = MaterialTheme.typography.subtitle1
                )

                ContentFacts(facts = plant.facts)

                plant.content.forEach { c ->
                    when {
                        c.type === ContentType.HEADER -> {
                            ContentHeader(header = c.content)
                        }
                        c.type === ContentType.TEXT -> {
                            ContentText(text = c.content)
                        }
                        c.type === ContentType.IMAGE -> {
                            ContentImage(url = c.content)
                        }
                        c.type === ContentType.MAP -> {
                            ContentMap(id = c.content)
                        }
                    }
                }

                FooterComponent(navController = navController)
            }
        }
    }
}